### PR TITLE
add make_tree_from_inorder_and_postorder.cpp/linked_list.cpp/linked_list.c

### PR DESCRIPTION
**Pull request by CanftIn**
**Fixes issue:**

**Changes:**

add make_tree_from_inorder_and_postorder.cpp/linked_list.cpp/linked_list.c

### CODE REVIEW

File: code/data_structures/src/linked_list/linked_list.cpp [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+#include <iterator>
+#include <sstream>
+#include <string>
+#include <list>
+#include <vector>
+
+
+class Node
+{
+public:
```
Comment (arnavb) [inline code/data_structures/src/linked_list/linked_list.cpp]: Typo?

File: code/data_structures/src/linked_list/linked_list.cpp [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+    int data;
+public:
+    Node(int data) : data(data) {}
+};
+
+class Linkedlist
+{
+private:
+    Node* head;
+public:
```
Comment (arnavb) [inline code/data_structures/src/linked_list/linked_list.cpp]: Public goes before private, simply for aesthetic reasons.

File: code/data_structures/src/linked_list/linked_list.cpp [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+    void insertatpos(int p, int data)
+    {
+  
+  //In case the list is empty, insert the value as the first element
+		if(head == nullptr)
+        {
```
Comment (arnavb) [inline code/data_structures/src/linked_list/linked_list.cpp]: I don't even want to know what happened to the indentation here.

File: code/data_structures/src/linked_list/linked_list.cpp [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+public:
+    Node(int data) : data(data) {}
+};
+
+class Linkedlist
+{
+private:
+    Node* head;
+public:
+    void insertatpos(int p, int data)
```
Comment (arnavb) [inline code/data_structures/src/linked_list/linked_list.cpp]: Names should be camel case.

File: code/data_structures/src/linked_list/linked_list.cpp [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+			std::cout << current->data << " ->";
+			current = current->next;
+		}
+	}
+};
+
+
+
+
+#endif
```
Comment (arnavb) [inline code/data_structures/src/linked_list/linked_list.cpp]: Need a newline here.

File: code/data_structures/src/linked_list/linked_list.cpp [hunk 6]
```diff
@@ -0,0 +1,115 @@
...
+			pos++;	
+			if(pos == p-1){
+				current->next = current->next->next;
+				return;
+			}
+			current = current->next;
+		}
+	}
+	
+    void display(){
```
Comment (arnavb) [inline code/data_structures/src/linked_list/linked_list.cpp]: Tsk. I'd prefer `std::ostream operator<<` implemented instead.

**Comment by CanftIn** [comment]
@arnavb I feel sorry and shamed about my work. I use space and tab all in my code, that's why make the code indentation so sick to review.
And I am very sincerely grateful for your patient review.
I have just rewrited all the code and add a simply test and format all code.

**Comment by arnavb** [comment]
Ok, sorry to have tell you this, but our guide says to use 4 spaces over tabs for indentation! (Just find a converter online). Also, I notice that you are using `std::list<Type> _elements` for storing the elements of the list. If I were to say this directly, doesn't this make the linked list pointless? If you are going to use a list internally to implement a linked list, then why not use the internal representation itself? It would be better if you used a series of connected pointers like in normal implementations.

**Comment by CanftIn** [comment]
@arnavb I used std::list inside LinkedList because I thought that's easy to extend some operator like `operator+/+=/[]` or map or more features that stl list does not exist.
Also, I've got what you mean, so I rewrited it in c just now(I don't want to change the cpp, just delete it). And I give a so typical implementation in cpp.
Of course, all tabs have been changed to 4 space.

File: code/data_structures/src/linked_list/linked_list.c [hunk 7]
```diff
@@ -0,0 +1,378 @@
...
+{
+    LinkedListNode* pHead;
+    LinkedListNode* pTail;
+    LinkedListNode* pCur;
+    UINT uCount;
+}LinkedList, *pLinkedList;
+
+
+LinkedList* LinkedList_Create();
+void LinkedList_Destroy(LinkedList* pLinkedList, DESTROYFUNC DestroyFunc);
```
Comment (arnavb) [inline code/data_structures/src/linked_list/linked_list.c]: 1. You need to follow our C style guide; Function  names should use snake_case (Fro the functional parts e.g `GetHead` -> `get_head`).
2. Other C style guide issues are also present, such as invalid placing of braces, return types going on their own line, etc. Please check the guide.

**Comment by CanftIn** [comment]
@arnavb Of course I need to do that. Now all have been updated.

**Comment by CanftIn** [comment]
@AdiChat Please check this pr, thanks.

**Comment by AdiChat** [comment]
Looks great 😍 

Hope you are enjoying your journey 🚆  with **OpenGenus**
positive_opengenus

**Pull request merged**